pages/shopping/index.js

```diff
@@ -124,54 +149,59 @@ export default function ShoppingList({ user, mutateUser }) {
     userShoppingList.splice(0, userShoppingList.length, ...nonEmptyCategories);
   }
 
-  consolidateShoppingListItems(user.shoppingList);
+  consolidateShoppingListItems(household.shoppingList);
 
   async function handleSubmit(event) {
     event.preventDefault();
     const formData = new FormData(event.target);
     const data = Object.fromEntries(formData);
     data.quantity = Number(data.quantity);
-    if (!user.shoppingList) {
-      user.shoppingList = [];
+    if (!household.shoppingList) {
+      household.shoppingList = [];
     }
 
-    const unsortedIndex = user.shoppingList.findIndex(
+    const unsortedIndex = household.shoppingList.findIndex(
       (category) => category.category === "Unsortiert"
     );
 
     if (unsortedIndex !== -1) {
-      user.shoppingList[unsortedIndex].items.push({
+      household.shoppingList[unsortedIndex].items.push({
         ...data,
         isChecked: false,
       });
     } else {
-      user.shoppingList.push({
+      household.shoppingList.push({
         name: "Unsortiert",
         items: [{ ...data, isChecked: false }],
       });
     }
 
-    await updateUserInDb(user, mutateUser);
+    updateHouseholdInDb(household, mutateHousehold);
     event.target.reset();
   }
 
   function handleCheckboxChange(categoryName, itemIndex) {
-    const categoryIndex = user.shoppingList.findIndex(
+    if (!userIsHouseholdAdmin) {
+      notifyError("Du besitzt keine Schreibrechte für diesen Haushalt.");
+      return;
+    }
+
+    const categoryIndex = household.shoppingList.findIndex(
       (c) => c.category === categoryName
     );
     if (categoryIndex === -1) {
       console.error("Kategorie nicht gefunden");
       return;
     }
 
-    const newUserShoppingList = [...user.shoppingList];
+    const newUserShoppingList = [...household.shoppingList];
     newUserShoppingList[categoryIndex].items[itemIndex].isChecked =
       !newUserShoppingList[categoryIndex].items[itemIndex].isChecked;
 
-    updateUserInDb(user, mutateUser);
+    updateHouseholdInDb(household, mutateHousehold);
 
     setTimeout(async () => {
-      const updatedCategories = [...user.shoppingList];
+      const updatedCategories = [...household.shoppingList];
```

Comment:
Please note that spreading only makes a shallow copy. That means if household.shoppingList contains references to other arrays or objects, then you would need to spread each of those as well to make a true copy. 

You can use something like [structuredClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) for deep copies.
(i think it needs to be node version 17 or above if you are using it server side).